cargo-dist/src/lib.rs

```diff
@@ -395,13 +396,133 @@ fn write_checksum(checksum: &str, src_path: &Utf8Path, dest_path: &Utf8Path) ->
     Ok(())
 }
 
+// Takes a string in KEY=value environment variable format and
+// parses it into a BTreeMap. The string syntax is sh-compatible, and also the
+// format returned by `env`.
+fn parse_env(env_string: &str) -> DistResult<BTreeMap<&str, &str>> {
+    let mut parsed = BTreeMap::new();
+    for line in env_string.trim_end().split('\n') {
+        let key;
+        let value;
+        match line.split_once('=') {
+            Some(parsed) => (key, value) = parsed,
+            None => {
+                return Err(DistError::EnvParseError {
+                    line: line.to_owned(),
+                })
+            }
+        };
+
+        parsed.insert(key, value);
```

Comment:
Is there established semantics for a key being defined multiple times? (This impl picks the last one)

Reply:
The string we're parsing from `env` will only ever return a given key once, so I'm trusting the input string to be well-formatted here. That said, I'll add a comment to that effect.